docs/app-containers.md

```diff
@@ -6,67 +6,53 @@ sidebar_label: App Containers
-
-A quick example of `createAppContainer`:
+A quick example of `NavigationContainer`:
 
 ```
-import { createAppContainer, createStackNavigator } from 'react-navigation';
-// you can also import from @react-navigation/native
-
-const AppNavigator = createStackNavigator(...);
-
-const AppContainer = createAppContainer(AppNavigator);
-
-// Now AppContainer is the main component for React to render
+import { NavigationContainer } from '@react-navigation/core';
+import { createStackNavigator } from '@react-navigation/stack';
+// you can also import NavigationContainer from @react-navigation/native
+
+const Stack = createStackNavigator();
+
+function App() {
+  return (
+    <NavigationContainer>
+      <Stack.Navigator>
+        { /* ... */ }
+      </StackNavigator>
+    </NavigationContainer>
+  )
+}
 
-export default AppContainer;
+export default App;
 ```
 
-## Props of `createAppContainer` on React Native
+## Props of `NavigationContainer` on React Native
 
 ```js
-<AppContainer
-  onNavigationStateChange={handleNavigationChange}
-  uriPrefix="/app"
+<NavigationContainer
+  onStateChange={handleStateChange}
+  initialState={initialState}
 />
 ```
 
-### `onNavigationStateChange(prevState, newState, action)`
+### `onStateChange(newState)`
 
-Function that gets called every time navigation state managed by the navigator changes. It receives the previous state, the new state of the navigation and the action that issued state change. By default it prints state changes to the console.
+Function that gets called every time navigation state managed by the navigator changes. It receives the new state of the navigation.
 
-### `uriPrefix`
+### `initialState`
 
-The prefix of the URIs that the app might handle. This will be used when handling a [deep link](deep-linking.html) to extract the path passed to the router.
+Prop that accepts initial state of navigation. You rather don't need for majority of cases but for deeplinking. #TODO
 
 ## Calling Dispatch or Navigate on App Container
 
-In case you want to dispatch actions on an app container, you can use a React [`ref`](https://facebook.github.io/react/docs/refs-and-the-dom.html#the-ref-callback-attribute) to call the `dispatch` method on it:
+In case you want to dispatch actions on an app container, you can use a React [`ref`](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) to call the `dispatch` method on it:
 
 ```js
-const AppContainer = createAppContainer(AppNavigator);
-
-class App extends React.Component {
-  someEvent() {
-    // call navigate for AppNavigator here:
-    this.navigator &&
-      this.navigator.dispatch(
-        NavigationActions.navigate({ routeName: someRouteName })
-      );
-  }
-  render() {
-    return (
-      <AppContainer
-        ref={nav => {
-          this.navigator = nav;
-        }}
-      />
-    );
-  }
+function App() {
+  const ref = useRef();
+  useEffect(() => ref.current.dispatch(CommonActions.goBack()));
```

Comment:
Probably not a good example to dispatch back in an effect which will trigger on first mount. Maybe add a button and do something `onPress` instead?